src/datanode/src/region_server.rs

```diff
@@ -285,12 +285,15 @@ impl RegionServerInner {
         // TODO(ruihang): add metrics and set trace id
 
         let QueryRequest {
-            header: _,
+            header,
             region_id,
             plan,
         } = request;
         let region_id = RegionId::from_u64(region_id);
 
+        let dbname = header.map(|h| h.dbname);
+        let ctx = QueryContext::with_db_name(dbname.as_ref());
```

Comment:
Perhaps we can serialize `QueryContext` into header directly, it will carry more context information automatically.

Reply:
`QueryContext` now have dynamic fields like `ArcSwap<Option<UserInfoRef>>` or `Box<dyn Dialect + Send + Sync>`, whom might be hard to be serialized. We can make a simplified version of `QueryContext`, but that would feel like another `header` too 🤔

Reply:
I see. Or if it's possible to add a `From<Header>` for QueryContext to formalize the conversion? The idea is to setup a connection between `QueryContext` in frontend and datanode, then make it maintainable when more fields added into the context

Reply:
https://github.com/GreptimeTeam/greptimedb/pull/2520/commits/81bd8dc9c2e7903d4182947b87e29aa3a62aa1b6
PTAL ☺️